server/py/framework/db/sqldb/db.py

```diff
@@ -1662,6 +1662,9 @@ def _find_artifacts(
         query = session.query(ArtifactV2).with_entities(
             ArtifactV2.id,
             ArtifactV2.Tag.name,
+            # Include tag ID (as 'tag_id') to enable sorting by tag creation order DESC.
+            # The alias is required to reference it later in subqueries and outer queries.
+            ArtifactV2.Tag.id.label("tag_id"),
```

Comment:
perhaps introduce a variable `tag_id_alias = "tag_id"` and use it in the `text` below. This will help connect the dots to whoever is reading this